src/ex1/scripts/domScripts.js

```diff
@@ -0,0 +1,195 @@
+  return removeTaskButton;
+}
+
+function addTaskDivEventListeners(taskDiv) {
+  taskDiv.addEventListener("click", onTaskClick);
+  taskDiv.addEventListener("mouseover", onTaskMouseOver);
+  taskDiv.addEventListener("mouseleave", onTaskMouseOut);
+  taskDiv.draggable = true;
+}
+
+function onTaskButtonsClick(e) {
+  const button = e.target;
+  if (button.classList.contains("task-complete-btn")) {
+    button.parentElement.classList.toggle("task-completed");
+    button.classList.toggle("btn-completed");
+    updateTaskInLocalStorage(
+      button.parentElement.querySelector(".task-item").innerText
+    );
+  } else if (e.target.classList.contains("task-remove-btn")) {
+    const task = e.target.parentElement;
+    task.addEventListener("transitionend", onRemoveTransitionEnd, false);
+    task.classList.add("task-removed");
+    function onRemoveTransitionEnd() {
+      removeTaskFromLocalStorage(task.innerText);
+      task.remove();
+      toggleEmptyMsg();
+      task.removeEventListener("transitionend", onRemoveTransitionEnd, false);
+    }
+  }
+}
+
+function onFilterOptionChange(e) {
+  const tasks = tasksList.children;
+  for (let i = 0; i < tasks.length; i++) {
+    const task = tasks[i];
+    const isCompleted = task.classList.contains("task-completed");
+    if (e.target.value === "all") {
+      task.style.display = "flex";
+    } else if (e.target.value === "completed" && isCompleted) {
+      task.style.display = "flex";
+    } else if (e.target.value === "uncompleted" && !isCompleted) {
+      task.style.display = "flex";
+    } else {
+      task.style.display = "none";
+    }
+  }
+}
+
+function toggleEmptyMsg() {
+  const emptyMsg = document.querySelector(".empty-msg");
+  if (tasksList.children.length === 0) {
+    emptyMsg.classList.remove("hide");
+  } else {
+    emptyMsg.classList.add("hide");
+  }
+}
+
+function onSearchInputKeyUp(e) {
+  const tasks = tasksList.children;
+  for (let i = 0; i < tasks.length; i++) {
+    const task = tasks[i];
+    if (task.innerText.toLowerCase().includes(e.target.value.toLowerCase())) {
+      task.style.display = "flex";
+    } else {
+      task.style.display = "none";
+    }
+  }
+}
+
+function onTaskClick(e) {
+  if (e.target.classList.contains("task")) {
+    alert(e.target.innerText);
+  }
+}
+
+function onTaskMouseOver(e) {
+  const completeTaskButton = e.target.querySelector(".task-complete-btn");
+  //because sometime the hover is on p and button is not the event
+  if (completeTaskButton != null) {
+    completeTaskButton.classList.remove("hide");
+    const removeTaskButton = e.target.querySelector(".task-remove-btn");
+    removeTaskButton.classList.remove("hide");
+  }
+}
+
+function onTaskMouseOut(e) {
+  const completeTaskButton = e.target.querySelector(".task-complete-btn");
+  completeTaskButton.classList.add("hide");
+  const removeTaskButton = e.target.querySelector(".task-remove-btn");
+  removeTaskButton.classList.add("hide");
+}
```

Comment:
You can achieve same behavior with css, no need to write JS for that.

example snippet for html + css to create a similar behavior: https://jsfiddle.net/Lpsedq8v/1/